src/lib/utils/template_type.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <boost/hana/type.hpp>
+
+namespace opossum {
+
+/**
+ * Wraps a class template so that
+ * it can be stored in a hana::type
```

Comment:
easy example?

Reply:
In `all_type_variant.cpp` we have something like:

```c++
auto data_types = make_tuple(make_pair(DataType::Int, type_c<int32_t>, ...));
```

… which we use to resolve data types. When resolving column types, we unfortunately can’t do the same, be cause a template class can’t simply be stored in `hana::type`. So this struct wraps template so that they can be stored in `hana::type`.